Core/include/Acts/AmbiguityResolution/ScoreBasedAmbiguityResolution.ipp

```diff
@@ -440,25 +402,194 @@ std::vector<int> Acts::ScoreBasedAmbiguityResolution::solveAmbiguity(
     trackScore = simpleScore(tracks, trackFeaturesVectors, optionalCuts);
   }
 
-  std::vector<bool> cleanTracks = getCleanedOutTracks(
-      trackScore, trackFeaturesVectors, measurementsPerTrack);
+  auto MeasurementIndexMap =
+      std::unordered_map<SourceLink, std::size_t, source_link_hash_t,
+                         source_link_equality_t>(0, sourceLinkHash,
+                                                 sourceLinkEquality);
+
+  std::vector<std::vector<std::size_t>> measurementsPerTrackVector;
+  std::map<std::size_t, std::size_t> nTracksPerMeasurement;
+
+  // Stores tracks measurement into a vector or vectors
+  // (measurementsPerTrackVector) and counts the number of tracks per
+  // measurement.(nTracksPerMeasurement)
+
+  for (const auto& track : tracks) {
+    std::vector<std::size_t> measurementsPerTrack;
+    for (const auto& ts : track.trackStatesReversed()) {
+      if (ts.typeFlags().test(Acts::TrackStateFlag::OutlierFlag) ||
+          ts.typeFlags().test(Acts::TrackStateFlag::MeasurementFlag)) {
+        ACTS_DEBUG("Track state type is OutlierFlag");
```

Comment:
this is not true. You also allow for measurements.

Reply:
Oh yes. that's a mistake, I will fix it.